Explorer/Controller/CreateLogViewController.swift

```diff
@@ -0,0 +1,134 @@
+    
+    var selectedDatePickerButton: UIButton?
+    
+    var diveDate = Date()
+    var diveStartTime = Date()
+    var diveEndTime = Date()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        scrollView.addSubview(createLogView)
+        view.addSubview(scrollView)
+        view.addSubview(datePickerView)
+        
+        datePickerView.doneButton.addTarget(self, action: #selector(setDate), for: .touchUpInside)
+        createLogView.diveDateButton.addTarget(self, action: #selector(presentPicker(button:)), for: .touchUpInside)
+        createLogView.startTimeButton.addTarget(self, action: #selector(presentPicker(button:)), for: .touchUpInside)
+        createLogView.endTimeButton.addTarget(self, action: #selector(presentPicker(button:)), for: .touchUpInside)
+        
+        createLogView.saveDiveButton.addTarget(self, action: #selector(saveDive), for: .touchUpInside)
+        
+        installConstraints()
+    }
+    
+    func installConstraints() {
+        scrollView.translatesAutoresizingMaskIntoConstraints = false
+        scrollView.topAnchor.constraint(equalTo: view.topAnchor).isActive = true
+        scrollView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
+        scrollView.leftAnchor.constraint(equalTo: view.leftAnchor).isActive = true
+        scrollView.rightAnchor.constraint(equalTo: view.rightAnchor).isActive = true
+        
+        createLogView.translatesAutoresizingMaskIntoConstraints = false
+        createLogView.topAnchor.constraint(equalTo: scrollView.topAnchor).isActive = true
+        createLogView.heightAnchor.constraint(equalToConstant: 1200).isActive = true
+        createLogView.centerXAnchor.constraint(equalTo: scrollView.centerXAnchor).isActive = true
+        createLogView.widthAnchor.constraint(equalTo: scrollView.widthAnchor).isActive = true
+        
+        // Date Picker view
+        datePickerView.translatesAutoresizingMaskIntoConstraints = false
+        datePickerBottomConstraint = datePickerView.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: 216)
+        datePickerBottomConstraint?.isActive = true
+        datePickerView.heightAnchor.constraint(equalToConstant: 216).isActive = true
+        datePickerView.leadingAnchor.constraint(equalTo: view.leadingAnchor).isActive = true
+        datePickerView.trailingAnchor.constraint(equalTo: view.trailingAnchor).isActive = true
+    }
+    
+    @objc func setDate() {
+        datePickerBottomConstraint?.constant = 216
+        
+        UIView.animate(withDuration: 0.34) {
+            self.view.layoutIfNeeded()
+        }
+        
+        switch selectedDatePickerButton {
+            case createLogView.diveDateButton:
+                diveDate = datePickerView.picker.date
+                let formatter = DateFormatter()
+                formatter.dateFormat = "dd MMM yyyy"
+                createLogView.diveDateButton.setTitle("  \(formatter.string(from: diveDate))", for: .normal)
```

Comment:
You have this line three times. You can pull the string out (ex. "  \(formatter.string(from: diveDate))") and create an empty string `var` on line 79 called `var buttonTitle = ""` and just update that depending on the case.

Then call `createLogView.diveDateButton.setTitle(buttonTitle, for: .normal)` at the end, down on line 97. Start looking for ways to consolidate code like this when you see yourself duplicating code like this.